mws/models/products.py

```diff
@@ -1,41 +1,76 @@
 """Datatype models for Products API."""
 
-from typing import Optional
-
-from mws.utils import flat_param_dict
+from enum import Enum
+from typing import Optional, Union
 
 from .base import MWSDataType
 
 
+class CurrencyCode(Enum):
+    """Constants for currency codes supported by Amazon."""
+
+    USD = ("USD", "United States dollar")
+    EUR = ("EUR", "European euro")
+    GBP = ("GBP", "Great Britain pounds")
+    RMB = ("RMB", "Chinese yuan")
+    INR = ("INR", "Indian rupee")
+    JPY = ("JPY", "Japanese yen")
+    CAD = ("CAD", "Canadian dollar")
+    MXN = ("MXN", "Mexican peso")
+
+    def __init__(self, code, description):
+        """Easy dot access like: Marketplaces.endpoint ."""
+        self.code = code
+        self.description = description
+
+    @property
+    def value(self):
+        return self.code
```

Comment:
Just want to call out, for these Enums being used in data models, having an override for `.value` property that returns the real piece of data required is a must.

Optionally, we could require each of these Enums implement some other property to do the same thing explicitly, but that's an extra step that might hinder further development, and I don't see the need for it.